### PR TITLE
CP-33636: add DCGM GPU metric collection and transformation

### PR DESCRIPTION
**Pull request by evan-cz**
CP-33636: add DCGM GPU metrics collection and transformation

Add complete NVIDIA DCGM GPU metrics support with Prometheus scraping
and transformation pipeline.

Prometheus Scrape Configuration:
- Added DCGM Exporter scrape job with Kubernetes service discovery
- Matches services labeled `app.kubernetes.io/name=dcgm-exporter`
- Collects DCGM_FI_DEV_GPU_UTIL, DCGM_FI_DEV_FB_USED, DCGM_FI_DEV_FB_FREE
- Includes 10 comprehensive Helm unit tests for scrape configuration

Metric Transformation Pipeline:
- Implemented hexagonal architecture for vendor-agnostic GPU metrics:
  - MetricTransformer interface in app/types (port)
  - Catalog transformer with sequential routing (domain service)
  - DCGM transformer for NVIDIA metrics (adapter)

DCGM Transformations:
- DCGM_FI_DEV_GPU_UTIL → container_resources_gpu_usage_percent
- DCGM_FI_DEV_FB_USED + FREE → container_resources_gpu_memory_usage_percent
- UUID label renamed to gpu_uuid with GPU- prefix stripped
- Supports both 'node' and 'Hostname' labels for node identification
- Memory metrics buffered and calculated as percentages in flush phase

Testing:
- 96.4% test coverage for DCGM transformer with table-driven tests
- Edge cases: missing labels, invalid values, incomplete pairs
- 10 Helm unit tests for scrape configuration and metric filtering
- Verified on EKS cluster (g4dn.xlarge with Tesla T4 GPU)

Documentation:
- README.md with transformation tables and processing strategy
- CLAUDE.md with AI development guide and architecture context
- Documented label transformations, node identification, and edge cases

Configuration:
- Added 100.0 to golangci-lint allowFloats for percentage conversion
- Updated Makefile for transform package test target

This provides foundation for multi-vendor GPU support (AMD, Intel)
following established architectural patterns in the codebase.

### CODE REVIEW

File: helm/templates/_cm_helpers.tpl [hunk 1]
```diff
@@ -163,6 +163,112 @@ remote_write:
...
+    - source_labels: [__name__]
+      regex: DCGM_FI_DEV_FB_USED
+      replacement: cz_gpu_memory_used_bytes
+      target_label: __name__
+
+    # Rename DCGM_FI_DEV_FB_FREE to cz_gpu_memory_free_bytes
+    - source_labels: [__name__]
+      regex: DCGM_FI_DEV_FB_FREE
+      replacement: cz_gpu_memory_free_bytes
+      target_label: __name__
```
Comment (dmepham) [inline helm/templates/_cm_helpers.tpl]: you mentioned different metrics for different vendors - would it make sense to add a label to the to differentiate? just so it might be easier to know how to process different metrics. just thinking out loud, I don't know if that would actually be helpful

**Review by dmepham — APPROVED** [review]
cool!

File: helm/templates/_cm_helpers.tpl (hunk 1 above)
Comment (evan-cz) [reply helm/templates/_cm_helpers.tpl]: I did! Sorry, I should have mentioned, but I add a provenance=dcgm label on lines 216-220:

```yaml
    # Add provenance label to indicate DCGM as the metric source
    - source_labels: [__meta_kubernetes_service_label_app_kubernetes_io_name]
      regex: dcgm-exporter
      replacement: dcgm
      target_label: provenance
```

I'm kind of wondering if we want to do some normalization in the collector, I don't love the thought of dealing with this complexity in the backend... That has some other benefits, too; I'll start a Slack thread in the morning.

Comment (dmepham) [reply helm/templates/_cm_helpers.tpl]: oh whoops, my bad, thanks!
